ddtrace/contrib/pytest/plugin.py

```diff
@@ -77,7 +78,16 @@ def pytest_configure(config):
     config.addinivalue_line("markers", "dd_tags(**kwargs): add tags to current span")
 
     if is_enabled(config):
-        Pin(tags=ci.tags(), _config=ddtrace.config.pytest).onto(config)
+        ci_tags = ci.tags()
+        if ci_tags.get(ci.git.REPOSITORY_URL, None):
+            try:
+                parsed_path = compat.parse.urlparse(ci_tags[ci.git.REPOSITORY_URL]).path
+                repository_name = parsed_path.split("/")[-1].split(".git")[0]
+            except Exception:
+                # In case of parsing error, default to repository url
+                repository_name = ci_tags[ci.git.REPOSITORY_URL]
```

Comment:
I'd like to see this refactored into it's own function so we can heavily test different url examples